### DISCUSSION TITLE
Adoptium ARM/no-ARM

### DISCUSSION THREAD

**Discussion by MikeLaptev** [paketo-buildpacks/feedback#287] [Java Team]
Could the table in the releases be extended with which SHA belongs which architecture?
https://github.com/paketo-buildpacks/adoptium/releases/tag/v11.5.0

at the same time, what is the process of selecting dedicated architecture, assuming that we have both options (`arm64` and `amd64`), so, where is the logic which define what JDK will be selected based on architecture?

**Comment by dmikusa** [comment]
> Could the table in the releases be extended with which SHA belongs which architecture? https://github.com/paketo-buildpacks/adoptium/releases/tag/v11.5.0

100% thanks for pointing this out.

https://github.com/paketo-buildpacks/pipeline-builder/issues/1629
 
> at the same time, what is the process of selecting dedicated architecture, assuming that we have both options (`arm64` and `amd64`), so, where is the logic which define what JDK will be selected based on architecture?

The buildpack makes the decision based on the current architecture. i.e. if you run on arm64 then it picks arm64 if you run on amd64 then it picks amd64 (the [logic for this is in libpak](https://github.com/paketo-buildpacks/libpak/blob/9895a3e241fddd5ba0454458a839b3744d3a261a/buildpack.go#L511-L518)). You can also set `BP_ARCH` if you want to force a particular architecture, but we really are only using that for test consistency.

**Reply by MikeLaptev** [reply]
> The buildpack makes the decision based on the current architecture. i.e. if you run on arm64 then it picks arm64 if you run on amd64 then it picks amd64 (the [logic for this is in libpak](https://github.com/paketo-buildpacks/libpak/blob/9895a3e241fddd5ba0454458a839b3744d3a261a/buildpack.go#L511-L518)).

Thank for clarifying this!
One more question, since the decision will happen at runtime, does it mean that the resulted image will contain both JDKs for `arm64` and `amd64`?

**Reply by dmikusa** [reply]
No, the decision happens when the buildpack runs, at build time. The application image produced will be architecture-specific. Presently, `pack` and Spring Boot Build Tools will only build app images for the architecture on which you run them (there's no cross-compile/build support). So if you run a build on ARM64, then it'll produce an ARM64-compatible container image. If you run a build on AMD64, then it'll produce an AMD64-compatible container image.

MacOS M-series hardware is an oddball here because it can technically do both. By default, it'll present as ARM64 though so you'll get ARM64 container images if you build on it. I don't think there's a way to force it to build AMD64 images, but it's possible that pack/Spring Boot Build Tools may let you force the platform in the future and if that were to happen then you could technically build both ARM64 and AMD64 images on this hardware.

**Reply by anthonydahanne** [reply]
there's a way to build for amd64 on Mac Silicon, even using dual arch buildpacks,  using `pack --platform`: [see this comment](https://github.com/paketo-buildpacks/java/discussions/1387#discussioncomment-9733781), last section

**Reply by dmikusa** [reply]
Yes, this is a new and great feature. Grab pack 0.34.0+.
